src/nupic/ntypes/BasicType.cpp

```diff
@@ -362,25 +363,27 @@ void BasicType::convertArray(void *ptr1, NTA_BasicType toType, const void *ptr2,
     case NTA_BasicType_Int32:
       switch (toType) {
       case NTA_BasicType_Byte:
-        cpyarray<Byte, Int32>(ptr1, ptr2, count, (Int32)std::numeric_limits<Byte>::min(), (Int32)std::numeric_limits<Byte>::max());
+        cpyarray<Byte, Int32>(ptr1, ptr2, count, static_cast<Int32>(std::numeric_limits<Byte>::min()), 
+                                                 static_cast<Int32>(std::numeric_limits<Byte>::max()));
```

Comment:
can these be left out w/o the static_cast? as those above

Reply:
> can these be left out w/o the static_cast? as those above

No, the Int32 has ranges both above and below that will not fit in a Byte (which is signed char).

Reply:
but std::numeric_limits<Byte>::max() would fit in Byte